echopype/echodata/simrad.py

```diff
@@ -118,11 +120,12 @@ def _retrieve_correct_beam_group_EK80(
     power_ed_group = None
     complex_ed_group = None
 
+    transmit_type_has_BB = not np.all(echodata["Sonar/Beam_group1"]["transmit_type"] == "CW")
```

Comment:
This is probably an expensive operation across all `ping_time`. Since right now we have this bug that we can only correctly process all-CW channels or all-BB channels (but not CW-BB mixed _within_ the same channel), do you think it's a bad idea to just test for `transmit_type` of the first ping?

Reply:
Surprisingly (at least to me), it's no more expensive than the `"CW" in dataarray` scheme we adopted elsewhere. See the tests I reported above, https://github.com/OSOceanAcoustics/echopype/pull/1091#issuecomment-1656841321

> Since right now we have this bug that we can only correctly process all-CW channels or all-BB channels

That's true, but the test is done across *all* channels, not for an individual channel. So, while the first ping tells us about the channel it corresponds to, it doesn't tell us anything about other channels.

Reply:
On second thought, maybe I wasn't seeing the multi-dimensional aspect of the data .... The "first" ping is the first ping in *each* channel. In that case, I guess you're right.

Reply:
I've implemented a solution based on the first ping_time.